karapace/schema_registry.py

```diff
@@ -145,130 +143,131 @@ async def get_master(self, ignore_readiness: bool = False) -> Tuple[bool, Option
-            subject_schemas_all = subject_data_all["schemas"]
+            schema_versions = self.subject_get(subject, include_deleted=True)
 
             # Subject can be permanently deleted if no schemas or all are soft deleted.
             can_permanent_delete = not bool(
-                [version for version, value in subject_schemas_all.items() if not value.get("deleted", False)]
+                [version_id for version_id, schema_version in schema_versions.items() if not schema_version.deleted]
             )
             if permanent and not can_permanent_delete:
                 raise SubjectNotSoftDeletedException()
 
             # Subject is soft deleted if all schemas in subject have deleted flag
-            already_soft_deleted = len(subject_schemas_all) == len(
-                [version for version, value in subject_schemas_all.items() if value.get("deleted", False)]
+            already_soft_deleted = len(schema_versions) == len(
+                [version_id for version_id, schema_version in schema_versions.items() if schema_version.deleted]
             )
-
             if not permanent and already_soft_deleted:
                 raise SubjectSoftDeletedException()
 
-            latest_schema_id = 0
+            latest_version_id = 0
             version_list = []
             if permanent:
-                version_list = list(subject_schemas_all)
-                latest_schema_id = version_list[-1]
-                for version, value in list(subject_schemas_all.items()):
-                    schema_id = value.get("id")
-                    LOG.info("Permanently deleting subject '%s' version %s (schema id=%s)", subject, version, schema_id)
+                version_list = list(schema_versions)
+                latest_version_id = version_list[-1]
+                for version_id, schema_version in list(schema_versions.items()):
+                    LOG.info(
+                        "Permanently deleting subject '%s' version %s (schema id=%s)",
+                        subject,
+                        version_id,
+                        schema_version.schema_id,
+                    )
                     self.send_schema_message(
-                        subject=subject, schema=None, schema_id=schema_id, version=version, deleted=True
+                        subject=subject, schema=None, schema_id=schema_version.schema_id, version=version_id, deleted=True
                     )
             else:
                 try:
-                    subject_data_live = self.subject_get(subject, include_deleted=False)
-                    version_list = list(subject_data_live["schemas"])
+                    schema_versions_live = self.subject_get(subject, include_deleted=False)
+                    version_list = list(schema_versions_live)
                     if version_list:
-                        latest_schema_id = version_list[-1]
+                        latest_version_id = version_list[-1]
```

Comment:
issue (non-blocking): Same thing here, we can reduce lines covered by the try-block. Not new from this PR, so feel free to ignore.